RecoVertex/PrimaryVertexProducer_Alpaka/plugins/alpaka/PortableTrackSoAProducer.cc

```diff
@@ -0,0 +1,244 @@
+#include "DataFormats/PortableVertex/interface/alpaka/VertexDeviceCollection.h"
+#include "DataFormats/PortableVertex/interface/VertexHostCollection.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/ESInputTag.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/global/EDProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDPutToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESGetToken.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/Math/interface/AlgebraicROOTObjects.h"
+
+#define DEBUG_RECOVERTEX_PRIMARYVERTEXPRODUCER_ALPAKA_PORTABLETRACKSOAPRODUCER 0
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  /**
+   * This:
+   * - consumes set of reco::Tracks and reco::BeamSpot
+   * - converts the reco::Tracks to a Alpaka-friendly dataformat portablevertex::TrackHostCollection
+   * - puts the Alpaka dataformat in the device for later consumption
+   */
+  struct filterParameters {
+    // Configurable filter parameters for the tracks
+    double maxSignificance;
+    double maxdxyError;
+    double maxdzError;
+    double minpAtIP;
+    double maxetaAtIP;
+    double maxchi2;
+    int minpixelHits;
+    int mintrackerHits;
+    reco::TrackBase::TrackQuality trackQuality;
+    double vertexSize;
+    double d0CutOff;
+  };
+
+  class PortableTrackSoAProducer : public global::EDProducer<> {
+  public:
+    PortableTrackSoAProducer(edm::ParameterSet const& config)
```

Comment:
The `config` needs to be passed to the base class now, along
```suggestion
    PortableTrackSoAProducer(edm::ParameterSet const& config): EDProducer(config)
```
I'd suggest to rebase to either 15_0_0_pre2 or pre3 first.